### ISSUE TITLE
Improve error message: ERROR: KeyError: key "uuid" not found

### ISSUE THREAD

**Issue by Datseris** [JuliaLang/Pkg.jl#491]
Hi, I tried to install this Julia package: https://github.com/JuliaMusic/MusicManipulations.jl

At the moment I tried to install it, the `Project.toml` was:
```
name = "MusicManipulations"
repo = "https://github.com/JuliaMusic/MusicManipulations.jl.git"
version = "0.1.0"

[deps]
MIDI = "f57c4921-e30c-5f49-b073-3f2f2ada663e"
Reexport = "189a3867-3050-52da-a836-e630ba90ab69"
Requires = "ae029012-a4dd-5104-9daa-d747884805df"
Statistics = "10745b16-79ce-11e8-11f9-7d13ad32a3b2"
```
And I got the error
```
(v0.7) pkg> dev https://github.com/JuliaMusic/MusicManipulations.jl
   Cloning git-repo `https://github.com/JuliaMusic/MusicManipulations.jl`
  Updating git-repo `https://github.com/JuliaMusic/MusicManipulations.jl`
ERROR: KeyError: key "uuid" not found
Stacktrace:
 [1] getindex at .\dict.jl:487 [inlined]
 [2] parse_package!(::Pkg.Types.Context, ::Pkg.Types.PackageSpec, ::String) at C:\cygwin\home\Administrator\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v0.7\Pkg\src\Types.jl:566
 [3] (::getfield(Pkg.Types, Symbol("##16#17")){Pkg.Types.Context,Array{Pkg.Types.PackageSpec,1}})(::LibGit2.CachedCredentials) at C:\cygwin\home\Administrator\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v0.7\Pkg\src\Types.jl:453
```
I think I am pretty sure the error was because the package did not have a `uuid` in the Project.toml, but I wasn't really sure how to correctly create one without breaking anything. It is not a registered package. 

Since the package already exists, I am not sure how to get a correct `uuid` (if I do `generate` in `pkg mode`, and copy the `uuid` from there, will it be ok?

**Comment by tpapp** [comment]
AFAIK `UUIDs.uuid1()` should work. You can also
```julia
shell> cd /tmp
pkg> generate MusicManipulations
```
and copy from `/tmp/MusicManipulations/Project.toml`; this is useful if you need the whole header.

**Comment by Datseris** [comment]
Yeap, `UUIDs.uuid1()` worked for me. I just pasted the content to a new `uuid` field in the Project.toml.

**Comment by fredrikekre** [comment]
Fixed by #476. On julia master:
```
(v0.7) pkg> dev Foo
ERROR: expected a `uuid` entry in project file at /home/fredrik/Foo/Project.toml
```
